contrib/format-syslog/src/main/java/org/apache/drill/exec/store/syslog/SyslogRecordReader.java

```diff
@@ -0,0 +1,395 @@
+
+    this.fileSystem = fileSystem;
+    this.fileWork = fileWork;
+    this.userName = userName;
+    this.config = config;
+    this.maxErrors = config.getMaxErrors();
+    this.df = getValidDateObject("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+    this.errorCount = 0;
+    this.buffer = context.getManagedBuffer(4096);
+    this.projectedColumns = columns;
+    this.flattenStructuredData = config.getFlattenStructuredData();
+
+
+    setColumns(columns);
+  }
+
+  public void setup(final OperatorContext context, final OutputMutator output) throws ExecutionSetupException {
+    openFile();
+    this.writer = new VectorContainerWriter(output);
+  }
+
+  private void openFile() {
+    InputStream in;
+    try {
+      in = fileSystem.open(new Path(fileWork.getPath()));
+    } catch (Exception e) {
+      throw UserException
+          .dataReadError(e)
+          .message("Failed to open open input file: %s", fileWork.getPath())
+          .addContext("User name", this.userName)
+          .build(logger);
+    }
+    this.lineCount = 0;
+    reader = new BufferedReader(new InputStreamReader(in, Charsets.UTF_8));
+  }
+
+  public int next() {
+    this.writer.allocate();
+    this.writer.reset();
+
+    int recordCount = 0;
+
+    try {
+      BaseWriter.MapWriter map = this.writer.rootAsMap();
+      String line = null;
+
+      while (recordCount < MAX_RECORDS_PER_BATCH && (line = this.reader.readLine()) != null) {
+        lineCount++;
+
+        // Skip empty lines
+        line = line.trim();
+        if (line.length() == 0) {
+          continue;
+        }
+        this.line = line;
+
+
+        try {
+          SyslogMessage parsedMessage = SyslogMessage.parseStructuredSyslogMessage(line);
+
+          this.writer.setPosition(recordCount);
+          map.start();
+
+          if (isStarQuery()) {
+            writeAllColumns(map, parsedMessage);
+          } else {
+            writeProjectedColumns(map, parsedMessage);
+          }
+          map.end();
+          recordCount++;
+
+        } catch (Exception e) {
+          errorCount++;
+          if (errorCount > maxErrors) {
+            throw UserException
+                .dataReadError()
+                .message("Maximum Error Threshold Exceeded: ")
+                .addContext("Line: " + lineCount)
+                .addContext(e.getMessage())
+                .build(logger);
+          }
+        }
+      }
+
+      this.writer.setValueCount(recordCount);
+      return recordCount;
+
+    } catch (final Exception e) {
+      errorCount++;
+      if (errorCount > maxErrors) {
+        throw UserException.dataReadError()
+            .message("Error parsing file")
+            .addContext(e.getMessage())
+            .build(logger);
+      }
+    }
+
+    return recordCount;
+  }
+
+  private void writeAllColumns(BaseWriter.MapWriter map, SyslogMessage parsedMessage) {
+    long milliseconds = parsedMessage.getTimestamp().getMillis(); //TODO put in try/catch
```

Comment:
It is theoretically possible that the dates in the Syslog file are not parsable or otherwise invalid and hence that method will throw exceptions.  Added an exception..